src/meta/processors/admin/SnapShot.cpp

```diff
@@ -88,19 +88,38 @@ cpp2::ErrorCode Snapshot::blockingWrites(storage::cpp2::EngineSignType sign) {
                    << static_cast<int32_t>(ret);
         return cpp2::ErrorCode::E_STORE_FAILURE;
     }
-    auto hosts = ActiveHostsMan::getActiveHosts(kv_);
-    for (const auto& host : hosts) {
-        std::for_each(spaces.begin(), spaces.end(), [host, sign, this](auto& space) {
+    for (auto& space : spaces) {
+        auto hostsBySpace = getHostsBySpace(space);
+        for (const auto& host : hostsBySpace) {
             auto status = client_->blockingWrites(space, sign, host).get();
             if (!status.ok()) {
                 LOG(ERROR) << " Send blocking sign error on host : "
                            << network::NetworkUtils::toHosts({host});
             }
-        });
+        }
     }
     return cpp2::ErrorCode::SUCCEEDED;
 }
 
+std::set<HostAddr> Snapshot::getHostsBySpace(GraphSpaceID space) {
+    folly::SharedMutex::ReadHolder rHolder(LockUtils::spaceLock());
+    std::set<HostAddr> hosts;
+    auto prefix = MetaServiceUtils::partPrefix(space);
+    std::unique_ptr<kvstore::KVIterator> iter;
+    auto kvRet = kv_->prefix(kDefaultSpaceId, kDefaultPartId, prefix, &iter);
+    if (kvRet != kvstore::ResultCode::SUCCEEDED) {
+        return {};
```

Comment:
maybe it is an error ?

Reply:
em... What's the difference between '{}' and 'empty list'?

Reply:
> maybe it is an error ?

I see what you mean. let me fix it.